src/prototype_creep_harvest.js

```diff
@@ -43,8 +43,18 @@ Creep.prototype.handleSourcer = function() {
 };
 
 Creep.prototype.spawnCarry = function() {
-  var energyThreshold = Game.rooms[this.memory.base].controller.level * config.sourcer.spawnCarryLevelMultiplier;
-  var waitTime = config.sourcer.spawnCarryWaitTime;
+  if (this.memory.wait > 0) {
+    this.memory.wait -= 1;
+    return false;
+  }
+
+  const foundKey = Object.keys(config.carry.sizes).reverse()
+    .find(key => (key <= Game.rooms[this.memory.base].energyCapacityAvailable));
+  let carryCapacity = config.carry.sizes[foundKey][1] * CARRY_CAPACITY;
+
+  const workParts = this.body.filter(part => part.type === WORK).length;
```

Comment:
I wouldn't rely on sorted body parts. If we change that at some point in time, it is really hard to find such issues.